openmc/lib/dagmc.py

```diff
@@ -0,0 +1,42 @@
+import sys
+from ctypes import c_int, c_int32, POINTER, c_size_t
+
+import numpy as np
+
+from . import _dll
+from .error import _error_handler
+
+
+# DAGMC functions
+_dll.openmc_get_dagmc_cell_ids.argtypes = [c_int32, POINTER(c_int32), POINTER(c_size_t)]
+_dll.openmc_get_dagmc_cell_ids.restype = c_int
+_dll.openmc_get_dagmc_cell_ids.errcheck = _error_handler
+
+
+def get_dagmc_cell_ids(volume_id, n_cells):
```

Comment:
it is possible, but we need the number of cells to instantiate the array to collect those ids to the right size, that means we need 2 calls to `_dll.openmc_get_dagmc_cell_ids(`
one to get the size, a second to fill the cell_ids array.

Passing a dynamic array did't work, so I had to use a fixed size array.


So what I can do:
```python
openmc.dagmc.sync_dagmc_cells() {
[...]
  cell_ids, found_cells = openmc.lib.dagmc.get_dagmc_cell_ids()
  assert found_cells == self.n_cells
[...]
}
```
with :
```python
openmc.lib.dagmc.get_dagmc_cell_ids(){
[...]
  # Get number of cells
  n = c_size_t()
  _dll.openmc_get_dagmc_cell_ids(volume_id, None, n)
  
  # Allocate array of cell ids 
  cell_ids = np.empty(n.value, dtype=np.int32)
  _dll.openmc_get_dagmc_cell_ids(volume_id, cell_ids.ctypes.data_as(POINTER(c_int32)), n )
[...]  
}

Reply:
Fair points, I'm thinking that it would probably make sense to have a separate CAPI call (something like `openmc_dagmc_universe_get_num_cells`) to get the number of cells from a DAGMC universe. The cross-over of information from the pure Python object to the Python object linked to a C++ object via `openmc.lib` is what feels off to me here.

Reply:
good idea !
I'll do that !